internal/controllers/rest_controller.go

```diff
@@ -70,14 +87,18 @@ func (rc RESTController) CheckForAuthentication(w http.ResponseWriter, r *http.R
 	// check if the token is expired or signature invalid.
 	// if so, the OIDC flow will be started to recreate a token.
 	if authErrors.IsTokenExpired(err) || authErrors.IsTokenSignatureInvalid(err) {
+		if authType == "HeaderCredentialsHandler" {
+			rc.forbidden(w, err)
+			c.Abort()
+			return
+		}
 		rc.logger.Warn(err)
 		rc.RedirectToOIDCProvider(w, r)
```

Comment:
we can't redirect when the client is a machine

Reply:
this is another case for us to think about it...